README.md

```diff
@@ -0,0 +1,150 @@
+# NimbleHex
+
+NimbleHex allows self-hosting of Hex packages.
+
+Features:
+
+* Pluggable storage. NimbleHex ships with local filesystem and S3 implementations
+* Publishing packages via HTTP API
+* Support for multiple repositories
+* Mirroring
+
+## Setup
+
+Clone and install dependencies:
+
+    git clone git@github.com:wojtekmach/nimble_hex.git
+    cd nimble_hex
+    mix deps.get
+
+Start a development server:
+
+    iex -S mix
+
+By default, the development server is configured with two repositories:
+
+1. `test_repo` is a custom repository
+2. `hexpm_mirror` is a mirror of the official Hex.pm repository, configured to only fetch package
+   "decimal".
+
+Both repositories are configured to store files locally. See `config/config.exs` and `config/dev.exs` for more information.
+
+Now, let's create a new package and publish it to our `test_repo` repository:
+
+    mix new foo
+    cd foo
+
+Make following changes to `mix.exs`:
+
+```elixir
+  def project() do
+    [
+      # ...
+      description: "Some description",
+      package: package(),
+      hex: hex()
+    ]
+  end
+
+  defp package() do
+    [
+      licenses: ["Apache-2.0"],
+      links: %{}
+    ]
+  end
+
+  defp hex() do
+    [
+      api_url: "http://localhost:4000/api/repos/test_repo",
+      api_key: "does-not-matter"
```

Comment:
oh, I was meaning to add a comment for it, will do that because it's confusing.

Without this, when the user is not authed already they'd get a prompt which will try to hit an http api endpoint that isn't implemented. And when the user happened to be already authed (e.g. against hex.pm/api), mix/hex would use that authed key but it's never checked as nimble_hex http api does not do any auth.

That's a good point that with this we'd additionally not leak api key to other servers.